### DISCUSSION TITLE
Format astro and svelte files in vscode

### DISCUSSION THREAD

**Discussion by joknoll** [denoland/deno#27173] [Q&A]
How can I configure vscode to format astro and svelte files with deno?

Currently I'm using the cli `deno fmt --unstable-components src/`, but having vscode autoformatting would be more ergonomic.

Thanks in advance!

**Comment by bartlomieju** [comment]
You need to have `deno.json` that enables `fmt-sql` unstable feature:
```json
{
  "unstable": [
    "fmt-sql"
   ]
}
```
then you'll need to specify this in your `.vscode/settings.json`:
```
{
  "editor.formatOnSave": true,
  "editor.defaultFormatter": "denoland.vscode-deno"
}
```

**Comment by joknoll** [comment]
Thanks for the answer. I've configured vscode and deno as described, but somehow it still doesn't work for me. Formatting SQL, Svelte or Astro with deno results in these popups.
However, when proceeding with "Configure", deno does not show up as an option.

![image](https://github.com/user-attachments/assets/48ac4268-9c81-466b-9b1b-4faa50ba17dc)

![image](https://github.com/user-attachments/assets/a5ca5426-79f2-446c-9dff-4bc885465ef8)

**Reply by bartlomieju** [reply]
Ooops, it should actually be `fmt-component` for Astro/Svelte files.

For the second part I'm gonna ping @nayeemrmn - are we not supporting it yet in `vscode_deno`?

**Reply by joknoll** [reply]
Just wanted to ask if there are any updates on this topic?
@nayeemrmn

**Comment by nanoz** [comment]
You could also selectively choose which formatter you want per extension.

Here is an example: https://github.com/KataLean/template-deno-astro/blob/main/.vscode/settings.json
